core/src/main/java/com/linecorp/armeria/server/ParameterizedPathMapping.java

```diff
@@ -148,12 +165,14 @@
      *   <li>{@code "{foo}"} -> {@code "foo"}</li>
      *   <li>{@code ":bar"} -> {@code "bar"}</li>
      *   <li>{@code "baz"} -> {@code null}</li>
+     *   <li>{@code "{*foo}"} -> {@code "foo"}</li>
      * </ul>
      */
     @Nullable
     private static String paramName(String token) {
         if (token.startsWith("{") && token.endsWith("}")) {
-            return token.substring(1, token.length() - 1);
+            final int beginIndex = token.indexOf('*') == 1 ? "{*".length() : "{".length();
```

Comment:
nit: I think using int value is much simpler and readable:
```suggestion
            final int beginIndex = token.indexOf('*') == 1 ? 2 : 1;
```